packages/node-core/src/indexer/poi/poi.service.ts

```diff
@@ -134,34 +135,46 @@ export class PoiService implements OnApplicationShutdown {
           queries.push(`ALTER TABLE ${tableName} ALTER COLUMN "chainBlockHash" DROP NOT NULL;`);
           // keep existing chainBlockHash
           queries.push(
-            `CREATE UNIQUE INDEX IF NOT EXISTS "poi_chainBlockHash" ON ${tableName} ("hash") WHERE "hash" IS NOT NULL`
+            sqlIterator(
+              tableName,
+              `CREATE UNIQUE INDEX IF NOT EXISTS "poi_chainBlockHash" ON ${tableName} ("hash") WHERE "hash" IS NOT NULL`
+            )
           );
         }
         if (!checkResult.hash_nullable) {
           queries.push(`ALTER TABLE ${tableName} ALTER COLUMN "hash" DROP NOT NULL;`);
-          queries.push(`UPDATE ${tableName} SET hash = NULL;`);
           queries.push(
-            `CREATE UNIQUE INDEX IF NOT EXISTS "poi_hash" ON ${tableName} ("hash") WHERE "hash" IS NOT NULL`
+            sqlIterator(
```

Comment:
Does iterators make sense with indexes?

Reply:
Yes, it would create index for any existing records. iterators make it much efficient and smooth. We did same for change index in dictionary project on host service.